h/accounts/models.py

```diff
@@ -132,6 +133,11 @@ def is_activated(self):
 
         return False
 
+    def activate(self, request):
+        """Activate the user by deleting any activation they have."""
+        request.db.delete(self.activation)
+        request.registry.notify(events.ActivationEvent(request, self))
```

Comment:
This needs to go somewhere else. SQLAlchemy model objects shouldn't be triggering Pyramid events, not least because they require the registry to do so. If that means that this one line is duplicated in both `admin.py` and `accounts/views.py` for now, then that's absolutely fine.